addons/point_of_sale/models/pos_session.py

```diff
@@ -308,19 +309,35 @@ def _validate_session(self):
             # Users without any accounting rights won't be able to create the journal entry. If this
             # case, switch to sudo for creation and posting.
             sudo = False
-            if (
-                not self.env['account.move'].check_access_rights('create', raise_exception=False)
-                and self.user_has_groups('point_of_sale.group_pos_user')
-            ):
-                sudo = True
-                self.sudo().with_company(self.company_id)._create_account_move()
-            else:
-                self.with_company(self.company_id)._create_account_move()
-            if self.move_id.line_ids:
-                # Set the uninvoiced orders' state to 'done'
-                self.env['pos.order'].search([('session_id', '=', self.id), ('state', '=', 'paid')]).write({'state': 'done'})
-            else:
-                self.move_id.unlink()
+            try:
+                if (
+                    not self.env['account.move'].check_access_rights('create', raise_exception=False)
+                    and self.user_has_groups('point_of_sale.group_pos_user')
+                ):
+                    sudo = True
+                    self.sudo().with_company(self.company_id)._create_account_move(balancing_account, amount_to_balance)
+                else:
+                    self.with_company(self.company_id)._create_account_move(balancing_account, amount_to_balance)
+                if self.move_id.line_ids:
+                    # Set the uninvoiced orders' state to 'done'
+                    self.env['pos.order'].search([('session_id', '=', self.id), ('state', '=', 'paid')]).write({'state': 'done'})
+                else:
+                    self.move_id.unlink()
+
+            except ImbalanceJournalEntryError as error:
+                # Creating the account move is just part of a big database transaction
+                # when closing a session. There are other database changes that will happen
+                # before attempting to create the account move, such as, creating the picking
+                # records.
+                # We don't, however, want them to be committed when the account move creation
+                # failed; therefore, we need to roll back this transaction before showing the
+                # close session wizard.
+                self.env.cr.rollback()
```

Comment:
Are you sure you don't want to put a `with self.env.cr.savepoint():` around the code in the `try:` clause instead?

Reply:
Thank you William for the review. Looks like the `savepoint` is the one I should have used. I'll try it.

Reply:
@william-andre @caburj remove the savepoint and set back the rollback, inventory is posted twice.

Reply:
No, it just mean that if we display the wizard, we want to rollback what have been done before. And making a savepoint at beginning of transaction to save nothing is useless.

Reply:
You don't know if it is at the beginning of a transaction. You might call that function in a more complex flow, in a server action or something.

Reply:
Even if it is not at the beginning of your transaction, you are not suppose to put savepoint at beginning of function without any good reason.

Just small example for your information: I override the function in another module, do something before calling super. This stuff will be committed (and it is not how it is suppose to work by default) And you'll get the same problem as with the stock posted multiple times.

If you really want to keep something before calling _validate_session, just call commit()

We won't change it, but feel free to keep those strange behaviors in your modules.